src/Domain.cpp

```diff
@@ -536,24 +536,28 @@ void Domain::writeCheckpointHeader(std::string filename,
 			for(auto pos=components->begin();pos!=components->end();++pos){
 				pos->write(checkpointfilestream);
 			}
-			unsigned int numperline=_simulation.getEnsemble()->getComponents()->size();
-			unsigned int iout=0;
-			for(auto pos=_mixcoeff.begin();pos!=_mixcoeff.end();++pos){
-				checkpointfilestream << *pos;
-				iout++;
-				// 2 parameters (xi and eta)
-				if(iout/2>=numperline) {
-					checkpointfilestream << std::endl;
-					iout=0;
-					--numperline;
-				}
-				else if(!(iout%2)) {
-					checkpointfilestream << "\t";
-				}
-				else {
-					checkpointfilestream << " ";
+			// Write mixing coefficients
+			auto mixingrules = _simulation.getEnsemble()->getMixingrules();
```

Comment:
```suggestion
			auto &mixingrules = _simulation.getEnsemble()->getMixingrules();
```
Avoid the unnecessary copy